Codecademy/gamut
# feat(LayoutMenuVariant): create layout menu variant for experiment #2199

### Overview

This pull request:
 - creates a LayoutMenuVariant for Disco to experiment with a new layout menu on the catalog page.
 - accomplishes this in a new variant component, instead of using too many conditionals in the existing component
 - changes include:  horizontal spacing of chevron, border to divide sections, font sizing, some vertical spacing between sections, and removal of selected behavior to match new design


See mocks for more details, but the LayoutMenu (existing) and new LayoutMenuVariant should look like this. 

_Existing_
<img width="328" alt="Screen Shot 2022-12-01 at 12 19 14 AM" src="https://user-images.githubusercontent.com/4481281/204971973-ac67e898-4008-4948-94f2-c4154f800ada.png">

_Variant_
<img width="286" alt="Screen Shot 2022-12-01 at 12 18 32 AM" src="https://user-images.githubusercontent.com/4481281/204972021-f6ec708b-1cc9-44f8-a405-2ffcf7fab936.png">





### PR Checklist

- [x] Related to designs: https://www.figma.com/file/rGzb9m1sPiFihBX3u8WEmG/Catalog-revamp?node-id=1288%3A45562
- [x] Related to JIRA ticket: [WEB-2199](https://codecademy.atlassian.net/browse/WEB-2199)

#### Testing Instructions

See new LayoutMenuVariant in the preview environment



[Don't make me tap the sign.](https://i.imgur.com/sy93D9I.png)

#### PR Links and Envs

| Repository   | PR Link                                                  | PR Env                                                   |
| :----------- | :------------------------------------------------------- | :------------------------------------------------------- |
| Monolith     | [Monolith PR](http://www.google.fr/ 'Named link title')  | [Monolith Env](http://www.google.fr/ 'Named link title') |
| Portal       | [Portal Link](http://www.google.fr/ 'Named link title')  | [Portal Env](http://www.google.fr/ 'Named link title')   |
| Another Repo | [Another Link](http://www.google.fr/ 'Named link title') | [Another Env](http://www.google.fr/ 'Named link title')  |
